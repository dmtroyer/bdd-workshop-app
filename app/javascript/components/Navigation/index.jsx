import React from 'react';
import {
  Collapse,
  Nav,
  NavItem,
  NavLink,
  Navbar,
  NavbarBrand,
} from 'reactstrap';
import { Link } from 'react-router-dom';

const Navigation = () => {
  return (
    <Navbar color="princess-pink" light expand="md">
      <NavbarBrand tag={Link} to="/" data-testid="landing-page-link">
        Game
      </NavbarBrand>
      <Collapse isOpen={true} navbar>
        <Nav navbar>
          <NavItem>
            <NavLink tag={Link} to="/about">
              About
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink tag={Link} to="/profile">
              Profile
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </Navbar>
  );
};

export default Navigation;
