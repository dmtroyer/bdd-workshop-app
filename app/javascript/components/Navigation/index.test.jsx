import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Navigation from '.';
import { MemoryRouter } from 'react-router-dom';

describe('Navigation', () => {
  afterEach(() => {
    cleanup();
  });

  it('Renders default unregistered view', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    // TODO actualy test for the Sign in and Register links
  });

  it('Renders signed in view', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation profile={{ profile: 'thing' }} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    // TODO actualy test for the Profile and Sign out links
  });
});
