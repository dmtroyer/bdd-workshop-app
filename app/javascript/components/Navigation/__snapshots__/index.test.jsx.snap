// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Renders default unregistered view 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-light bg-princess-pink"
  >
    <a
      class="navbar-brand"
      data-testid="landing-page-link"
      href="/"
    >
      Game
    </a>
    <button
      class="mr-2 navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
    >
      <ul
        class="rightNav ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="sign-in-link"
            href="/sign_in"
          >
            Sign in
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="register-link"
            href="/register"
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`Navigation Renders signed in view 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-light bg-princess-pink"
  >
    <a
      class="navbar-brand"
      data-testid="landing-page-link"
      href="/"
    >
      Game
    </a>
    <button
      class="mr-2 navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
    >
      <ul
        class="rightNav ml-auto navbar-nav"
      >
        <li
          class="dropdown nav-item"
          data-testid="details"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            data-testid="summary"
            href="#"
          >
            <i
              alt="placeholder-avatar"
              aria-hidden="true"
              class="fa fa-user"
              data-testid="details-avatar"
            />
            <span>
               
            </span>
          </a>
          <div
            aria-hidden="true"
            class="dropdown-menu dropdown-menu-right"
            role="menu"
            tabindex="-1"
          >
            <a
              class="dropdown-item"
              data-testid="profie-link"
              href="/profile"
              role="menuitem"
              tabindex="0"
            >
              Profile
            </a>
            <a
              class="dropdown-item"
              href="/sign_out"
              role="menuitem"
              tabindex="0"
            >
              Sign out
            </a>
          </div>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
